### DISCUSSION TITLE
Extremely hard to configure forge to use with github enterprise

### DISCUSSION THREAD

**Discussion by jbulow** [magit/forge#649] [Support] [answered]
Spent many hours trying to configure magit/forge to work with GHE. The error just say: (anonymized URL)

```
Cannot determine forge repository.  git@github.mycompany.com:org/repo.git isn’t a forge url
```

No hint on how to fix it. Read the manual several times (that seems to be more of a developers guide). Could not find any useful examples in the manual. In the end, it seems like a few lines showing how to load forge with `use-package` and the necessary configuration and initialization should help a lot. 

Normal magit works fine with my GHE instance. Not relevant here, but I have also configured the github CLI tool to use the GHE instance. 

I just fail to understand why the manual is so vague about how to setup forge.

**Comment by tarsius** [comment, marked as answer]
This seems like an unfair and inaccurate description of the existing documentation.

> ```
> Cannot determine forge repository.  git@github.mycompany.com:org/repo.git isn’t a forge url
> ```
>
> No hint on how to fix it.

That part is fair enough. I'll probably extend that to mention that this likely means that `forge-alist` lacks a corresponding entry. Do `C-h v forge-alist RET` to learn more about that variable, look at the existing entries, and add an entry for your github instance. 

Note in particular the complication about Github enterprise instances at the end of the docstring. I'll probably later add an example to that.
